wiki/python/ESTIMATOR_GUIDE.md

```diff
@@ -0,0 +1,866 @@
+# cuML Python Estimators Developer Guide
+
+This guide is meant to help developers follow the correct patterns when creating/modifying any cuML Estimator object and ensure a uniform cuML API.
+
+**Note:** This guide is long, because it includes internal details on how cuML manages input and output types for advanced use cases. But for the vast majority of estimators, the requirements are very simple and can follow the example patterns shown below in the [Quick Start Guide](#quick-start-guide).
+
+Start by reading the following:
```

Comment:
If this is required as part of the quick start guide then I would move it there, otherwise I would move it to just after the quick start guide. Otherwise there's already a _lot_ of information here presented before the quick start guide which might cost a reader that doesn't need it significant time.

Reply:
Would be good to discuss this offline. I think we feel differently about how much information is before the quick start guide. Moving it after the guide seems awkward to me.

Reply:
I get Dante's point and have generally supported getting the overview near the start, but I don't think this is a problem to leave in as it is important background.

Reply:
Talked with Dante. Going to change it from "required" reading to "suggested"